pandas/tests/groupby/test_apply.py

```diff
@@ -851,3 +851,16 @@ def test_apply_function_returns_non_pandas_non_scalar(function, expected_values)
     result = df.groupby("groups").apply(function)
     expected = pd.Series(expected_values, index=pd.Index(["A", "B"], name="groups"))
     tm.assert_series_equal(result, expected)
+
+
+def test_type_error_complex_index_non_scalar_return():
+    # GH 31605
+    def fct(group):
+        return group[1].values.flatten()
+
+    df = pd.DataFrame([("A", 1), ("A", 2), ("A", 3), ("B", 4), ("B", 5), ("C", np.nan)])
+    result = df.groupby(0).apply(fct)
```

Comment:
I know it's the example of the bug report, but can you give the dataframe column names? (eg ['a', 'b'])

Reply:
done